src/fields2cover/path_planning/path_planning.cpp

```diff
@@ -125,11 +120,23 @@ std::vector<std::pair<F2CPoint, double>> PathPlanning::simplifyConnection(
     }
   }
 
+  if (p1.distance(p2) < 6.0 * safe_dist || vp.size() <2) {
```

Comment:
Why moving section this down? Early returns are prefered

Reply:
Im not really moving it down. Rather, I'm defining `vp` earlier and using the same early return but using `vp` instead of `mp`.

Reply:
I suppose it is going through the for loop to create `vp` each time. We could just add a second check for if `vp.size() < 2` that uses the same return value.